modules/cli/src/main/scala/scala/cli/commands/SharedCompilationServerOptions.scala

```diff
@@ -160,14 +163,38 @@ final case class SharedCompilationServerOptions(
   def minimumBloopVersion = Constants.bloopVersion
   def acceptBloopVersion  = Some((v: String) => Version(v) < Version(minimumBloopVersion))
 
+  case class BloopJson(javaHome: String, javaOptions: Array[String])
+
+  def bloopDefaultJvmOptions(logger: Logger): List[String] = {
+    val file = new File(bloopGlobalOptionsFile)
+    if (file.exists() && file.isFile()) {
+      try {
+        val reader = new BufferedReader(new FileReader(file))
+        val gson   = new Gson()
```

Comment:
When using gson, you should add the class you're deserializing to `reflect-config.json` for it to work from the native launchers. See [how it's done for `BspConnectionDetails`](https://github.com/VirtusLab/scala-cli/blob/810586715d4693176d3ec49a7ef4904e5b90df76/modules/cli/src/main/resources/META-INF/native-image/org.virtuslab/scala-cli-core/reflect-config.json#L2-L8).